src/formatDate.js

```diff
@@ -50,7 +50,27 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArr = date.split(fromFormat[fromFormat.length - 1]);
+  const dateObj = {};
+  const resultDateFormat = [];
+
+  for (let i = 0; i < dateArr.length; i++) {
+    if (fromFormat[i] === 'YY') {
+      dateArr[i] = dateArr[i] < 30 ? '20' + dateArr[i] : '19' + dateArr[i];
+      fromFormat[i] = 'YYYY';
+    }
+    dateObj[fromFormat[i]] = dateArr[i];
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    if (toFormat[i] === 'YY') {
+      resultDateFormat.push(dateObj.YYYY.slice(2));
+    } else {
+      resultDateFormat.push(dateObj[toFormat[i]]);
+    }
```

Comment:
here you can apply ternary as well
```suggestion
    resultDateFormat.push(toFormat[i] === 'YY' ? dateObj.YYYY.slice(2) : dateObj[toFormat[i]]);
```